tests/test_edgeql_casts.py

```diff
@@ -2162,6 +2162,50 @@ async def test_edgeql_casts_json_10(self):
             await self.con.query_one(
                 'SELECT <cal::local_time><json>"20:01:22.306916+01:00"')
 
+    async def test_edgeql_casts_json_11(self):
+        await self.assert_query_result(
+            r"SELECT <array<int64>><json>[1, 1, 2, 3, 5]",
+            [[1, 1, 2, 3, 5]]
+        )
+
+        with self.assertRaisesRegex(
+                # FIXME: This should be a different error
+                edgedb.InternalServerError,
+                r'expected json number, null; got json string'):
+            await self.con.query_one(
+                r"SELECT <array<int64>><json>['asdf']")
+
+        with self.assertRaisesRegex(
+                # FIXME: This should be a different error
+                edgedb.InternalServerError,
+                r'expected json number, null; got json string'):
```

Comment:
The error message is somewhat misleading. I wonder if we can simply avoid mentioning `null` in `jsonb_assert_type`.

Reply:
This is material for a followup PR, though.